err.go

```diff
@@ -29,10 +33,19 @@ func (e *InvalidOperationError) Error() string {
 
 // InvalidArgumentError is returned when one of the arguments given to a function was not valid.
 type InvalidArgumentError struct {
-	argumentName string
+	argumentName  string
 	argumentValue interface{}
 }
 
 func (e *InvalidArgumentError) Error() string {
 	return fmt.Sprintf(`ionhash: invalid value: "%v" specified for argument: %s`, e.argumentValue, e.argumentName)
 }
+
+// An InvalidIonTypeError is returned when processing an unexpected ion type.
+type InvalidIonTypeError struct {
+	ionType ion.Type
+}
+
+func (e *InvalidIonTypeError) Error() string {
+	return fmt.Sprintf(`ionhash: invalid ion type: %s`, e.ionType.String())
```

Comment:
nit: Ion

Reply:
I only went with lower case i because the other error types all have lower case for their error messages. I will adjust the other error types as well.

Reply:
the package name `ionhash` should be lower case but `ion` should be capitalized

Reply:
Fixed.